CODE_CONVENTIONS.md

```diff
@@ -97,3 +98,108 @@ stop = stop.copyOf().withPrivateCode("TEX").build();
 #### Updaters run with a context(editor)
 
 TODO See issue 4002 - Document when implemented
+
+
+## Records, POJOs and Builders
+
+We prefer immutable typesafe types over flexibility and "short" class definitions. This make
+the code more robust and error prune.
+
+### Recods
+
+You may use records, but avoid using records if you can not encapsulate it properly. Be especially
+aware of arrays fields (can not be protected) and collections (remember to make a defensive copy).
+If you need to override `equals` and `hashCode`, then it is probably not worth it.
+Be aware that `equals` compare references, not the value of a field. Consider overriding `toString`.
+
+### Builders
+
+OTP used a simple builder pattern in many places, especially when creating immutable types.
+
+#### Builder conventions
+- Use factory methods to create builder, either `of()` or `copyOf()`. The _copyOf_ uses an existing
+  instance as its base. The `of()` creates a builder with all default values set. All constructors 
+  should be private (or package local) to enforce the use of the factory methods.
+- If the class have more than 5 fields avoid using an inner class builder, instead create a builder
+  in the same package.
+- Validate all fields in the constructor, especially null checks. Prefer default values over
```

Comment:
Possibly clarify where the validation takes place: Validate all fields in the constructor --> Validate all fields in the type constructor (i.e not in the builder)
Possibly insist on the use of final field to express intent on immutablity
Possibly mention utility methods for checking parameters  (Objects.requireNonNull, ...)